src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('DELETE', 'api/clients/'.self::UNEXISTING_ID);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.self::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClients(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients');
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+        '@context' => '/api/contexts/Client',
+        '@type' => 'hydra:Collection',
+        '@id' => '/api/clients',
+        ]);
+    }
+
+    public function testGetClientsUnauthenticated(): void
+    {
+        $response = static::createClient()->request('GET', '/api/clients');
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetNonexistentClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('GET', '/api/clients/'.self::UNEXISTING_ID);
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertHydraError('The client "'.self::UNEXISTING_ID.'" does not exist.');
+    }
+
+    public function testGetClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => $clientName
+        ]);
+        $response = $httpClient->request('GET', $iri);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testUpdateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $iri = $this->findIriBy(Client::class, ['name' => 'client_2']);
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
+        $updatedName = $this->createClientName();
+        $updateClientJson = ClientMother::withAllParameters(
+            $updatedName,
+            1,
+            "description updated",
+            true,
+            5,
+            [$scriptId],
+            [],
+            -1,
+            "rsync long arguments updated",
+            "rsync short arguments updated",
+            "ssh arguments updated",
+            "root@172.0.0.2"
+        );
+        $httpClient->request('PUT', $iri, [
+            'json' => $updateClientJson
+        ]);
```

Comment:
If you execute the test twice the second time client_2 won't exist unless database is reset and fixtures reloaded. This makes this quite fragile and not very convinient. A test should be as autosufficient as posible and so fixtured or preloaded data should not be changed.

I suggest something like:

$client = ClientMother::random() 
// do the post and assertions
$updated = ClientMother:random()
// Post to same iri the updated client and assert